tests/unit/test_state.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Jenkins-k8s state module tests."""
+
+from unittest.mock import MagicMock
+
+import pytest
+
+import state
+
+
+@pytest.mark.parametrize(
+    "time_range",
+    [
+        pytest.param("-1-3", id="negative time range"),
```

Comment:
Can you test for an invalid format too?

Reply:
These are all invalid test cases :)

Reply:
I wasn't very clear :D I meant a string not matching the H(H)-H(H) pattern

Reply:
Added the extra tests, thank you for the review!